qulice-checkstyle/src/main/java/com/qulice/checkstyle/JavadocEmptyLineCheck.java

```diff
@@ -0,0 +1,136 @@
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.qulice.checkstyle;
+
+import com.puppycrawl.tools.checkstyle.api.Check;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Check for empty line at the start and at the end of javadoc.
+ * @author Denys Skalenko (d.skalenko@gmail.com)
+ * @version $Id$
+ * @since 0.17
+ */
+public final class JavadocEmptyLineCheck extends Check {
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.CLASS_DEF,
+            TokenTypes.INTERFACE_DEF,
+            TokenTypes.VARIABLE_DEF,
+            TokenTypes.CTOR_DEF,
+            TokenTypes.METHOD_DEF,
+        };
+    }
+
+    @Override
+    public void visitToken(final DetailAST ast) {
+        final String[] lines = this.getLines();
+        final int start = ast.getLineNo();
+        final int cstart =
+            JavadocEmptyLineCheck.findCommentStart(lines, start) + 1;
+        if (JavadocEmptyLineCheck.isNodeHasJavaDoc(ast, cstart)) {
+            if (JavadocEmptyLineCheck.isJavadocLineEmpty(lines[cstart])) {
+                this.log(cstart + 1, "Empty javadoc line at the beginning");
+            }
+            final int cend =
+                JavadocEmptyLineCheck.findCommentEnd(lines, start) - 1;
+            if (JavadocEmptyLineCheck.isJavadocLineEmpty(lines[cend])) {
+                this.log(cend + 1, "Empty javadoc line at the end");
+            }
+        }
+    }
+
+    /**
+     * Check if javadoc Line Empty.
+     * @param line Java doc line
+     * @return Is Javadoc Line Empty
+     */
+    private static boolean isJavadocLineEmpty(final String line) {
+        return null != line && "*".equals(line.trim());
+    }
+
+    /**
+     * Check if node has java doc.
+     * @param node Node to be checked for Java docs.
+     * @param start Line number where comment starts.
+     * @return Is node has java doc
+     */
+    private static boolean isNodeHasJavaDoc(final DetailAST node,
+        final int start) {
+        int cprevious = 0;
+        final DetailAST previous = node.getPreviousSibling();
+        if (null != previous) {
+            cprevious = previous.getLineNo();
+        }
+        return start > cprevious;
+    }
+
+    /**
+     * Find javadoc starting comment.
```

Comment:
@dskalenko please be consistent and always write Javadoc word as `Javadoc`, not `javadoc` or `JavaDoc`